pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -196,6 +215,24 @@ func createDynamicMetadataMatcher(k, v string) *metadata.MetadataMatcher {
 		keys = []string{k}
 	}
 
+	var listMatcher *metadata.ListMatcher
+	if k == attrRequestGroups {
```

Comment:
could you merge the `if k == attrRequestGroups` to the `if` statements starting from line 201?